lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
```

Comment:
@domenic One more problem: it looks like `resolver` now uses `resolve` and `reject` which are out of it's scope (outside of the `Promise` closure).